app/services/service_container.rb

```diff
@@ -0,0 +1,43 @@
+require "dry/container"
+
+class ServiceContainer
+  def initialize
+    @services = Dry::Container.new
+    register_services
+  end
+
+  def define(namespace, name, &block)
+    @services.namespace(namespace) do
+      register(name) { block.call }
```

Comment:
Couldn't you pass `block` through to `register` here? Or am I missing something ...

Reply:
Yes, apparently I just forgot how blocks work :upside_down_face: